DietPi-Software | O!MPD: Update download URL

Joulinar opened the pull request:
@MichaIng 

Main web site for O!MPD as been taken down due to copyright violations. Therefore we get `HTTP 451` while checking the URL

`HTTP request sent, awaiting response... 451 Unavailable for Legal Reasons `

However direct download is still working. Therefore it's needed to check against the download file instead of main web site.

**Status**: Ready
- [x] update download URL

**Reference**: 
- https://dietpi.com/phpbb/viewtopic.php?f=11&t=8242
- https://youtube-dl.org/
- https://github.com/github/dmca/blob/master/2020/10/2020-10-23-RIAA.md

**Commit list/description**:
- DietPi-Software | O!MPD: Update download URL as main web site has been taken down

Commit:
Update dietpi-software

DietPi-Software | O!MPD: Update download link as main web site has been taken down

MichaIng commented:
Ah great, I think in the past the direct download couldn't be checked, as usual for files pulled from GitHub releases. The link obviously has been changed since `youtube-dl` has been taken down from GitHub, hence the direct link now can be checked.

Actually, I think we should remove `youtube-dl` completely until things have been settled the one or the other way. When they are allowed on GitHub again, your change will not work anymore (like before) and when their website/domain is taken down completely, it will not work anymore at all, of course.

Last but not least we generally should not automatically download a tool for our users, which is currently in discussion for being illegal. One can always download it manually, but as long as the legal status is not clear, it should be not our decision. This is not about a personal opinion or the discussion whether a tool which can be used for illegal activity should automatically be illegal as well, but simply to wait for official decisions/processes about that case to not risk the own legal state and, more importantly, the one of our users.

Joulinar commented:
yep was thinking the same to have it disabled for time being. Maybe that's indeed better option as of now.

MichaIng commented:
Had another review. Actually it's only GitHub which removed it for being no compliant with DCMA. The releases link is just redirected to the blocked GitHub releases page.

`youtube-dl` does not only allow to download from YouTube but a lot of other sites with often or only legally downloadable content. Of course it can be misused, but with that logic alone one would need to ban all browsers as well.
I think we can take Debian as reference and it did not remove `youtube-dl` from it's repository: https://packages.debian.org/youtube-dl

Just in case it is unblocked on GitHub and the direct download is redirected to GitHub again, I'll remove the separate `G_CHECK_URL`. It is anyway somehow doubled as `curl` itself will throw the same error messages anyway. I plan to place `G_CHECK_URL` with `G_DOWNLOAD` to show some targeted solution attempts from the error handler, have a consistent method and limit DNS resolving + connection times. `G_CHECK_URL` still has some use, but not for downloads.

Joulinar commented:
make sense to combine URL checking as well as download into a single option. 👍

MichaIng committed:
v6.34

+ DietPi-Software | O!MPD: Skip G_CHECK_URL for youtube-dl download, in case it will be redirected to GitHub again.
+ DietPi-Software | O!MPD: Handle extracted OMPD-master dir explicitly and use G_EXEC where applicable

MichaIng committed:
v6.34

+ CHANGELOG | O!MPD: Resolved an issue where the URL check for youtube-dl failed.

Joulinar commented:
@MichaIng 
just as information https://www.heise.de/news/Nach-Abmahnung-durch-die-Musikindustrie-Youtube-DL-ist-wieder-da-4963082.html

it's unblocked on GitHub again

MichaIng commented:
Especially nice to read the GitHub aims to defend developers better in similar cases and work together with EFF (Certbot developers btw) and SFLC in this case.

However, nothing to chance back our side as it's fine to download the script directly like we do.
